py/src/nupic/examples/tm/tm_overlapping_sequences.py

```diff
@@ -0,0 +1,917 @@
+    print("\nTraining sequences")
+    printAllTrainingSequences(trainingSequences)
+
+  return (numCols, trainingSequences)
+
+
+
+def buildSequencePool(numSequences = 10,
+                      seqLen = [2,3,4],
+                      numPatterns = 5,
+                      numOnBitsPerPattern = 3,
+                      patternOverlap = 0,
+                      **kwargs
+                      ):
+  """ Create a bunch of sequences of various lengths, all built from
+  a fixed set of patterns.
+
+  Parameters:
+  -----------------------------------------------------
+  numSequences:         Number of training sequences to generate
+  seqLen:               List of possible sequence lengths
+  numPatterns:          How many possible patterns there are to use within
+                          sequences
+  numOnBitsPerPattern:  Number of ON bits in each TM input pattern
+  patternOverlap:       Max number of bits of overlap between any 2 patterns
+  retval:               (numCols, trainingSequences)
+                          numCols - width of the patterns
+                          trainingSequences - a list of training sequences
+
+  """
+
+
+  # Create the table of patterns
+  patterns = getSimplePatterns(numOnBitsPerPattern, numPatterns, patternOverlap)
+
+  # Total number of columns required
+  numCols = len(patterns[0])
+
+
+  # -----------------------------------------------------------------------
+  # Create the training sequences
+  trainingSequences = []
+  for _ in range(numSequences):
+
+    # Build it up from patterns
+    sequence = []
+    length = random.choice(seqLen)
+    for _ in range(length):
+      patIdx = random.choice(range(numPatterns))
+      sequence.append(patterns[patIdx])
+
+    # Put it in
+    trainingSequences.append(sequence)
+
+
+  if VERBOSITY >= 3:
+    print("\nTraining sequences")
+    printAllTrainingSequences(trainingSequences)
+
+  return (numCols, trainingSequences)
+
+
+
+def createTMs(includeCPP = True,
+              includePy = True,
+              numCols = 100,
+              cellsPerCol = 4,
+              activationThreshold = 3,
+              minThreshold = 3,
+              newSynapseCount = 3,
+              initialPerm = 0.6,
+              permanenceInc = 0.1,
+              permanenceDec = 0.0,
+              globalDecay = 0.0,
+              pamLength = 0,
+              checkSynapseConsistency = True,
+              maxInfBacktrack = 0,
+              maxLrnBacktrack = 0,
+              **kwargs
+              ):
+
+  """Create one or more TM instances, placing each into a dict keyed by
+  name.
+
+  Parameters:
+  ------------------------------------------------------------------
+  retval:   tms - dict of TM instances
+  """
+
+  # Keep these fixed:
+  connectedPerm = 0.5
+
+  tms = dict()
+
+  if includeCPP:
+    if VERBOSITY >= 2:
+      print("Creating BacktrackingTMCPP instance")
+
+    cpp_tm = BacktrackingTMCPP(numberOfCols = numCols, cellsPerColumn = cellsPerCol,
```

Comment:
Again, the C++ part of BacktrackingTMCPP is gone.

Reply:
Removed this file.